addons/udes_stock/models/stock_move.py

```diff
@@ -229,12 +293,74 @@ def _create_split_move(self, move_lines, remaining_qty, new_move_qty, **kwargs):
 
         # When not complete, splitting a move may change its state,
         # so recompute
-        incomplete_moves = (self | new_move).filtered(lambda mv: mv.state not in ["done", "cancel"])
+        incomplete_moves = (self | new_move).get_incomplete_moves()
         incomplete_moves._recompute_state()
 
         move_lines.write({"state": new_move.state})
 
         if self.move_orig_ids:
-            (new_move | self).update_orig_ids(self.move_orig_ids)
+            # Update the current and new move original move ids.
+            (self | new_move).update_orig_ids(self.move_orig_ids)
 
         return new_move
+
+    def _make_mls_comparison_lambda(self, move_line):
+        """
+        This makes the lambda for checking the a move_line against move_orign_ids
+        This can be overridden in other modules if desired.
+        If no matching is made, it defaults to a matching on the dest location
+        and product id. This is not very robust and can lead to over-matching.
+        
+        The saving grace of the default behaviour is that it is not very common.
+        Normally, if not lot tracked, there are packages/pallets, and if not there
+        are procurment groups which block the merging of moves/pickings.
+
+        :args:
+            - move_line to match self with
+        :returns:
+            - lambda expression for mathcing
+        """
+        lot_name = move_line.lot_id.name or move_line.lot_name
+        package = move_line.package_id
+
+        if lot_name and package:
+            return lambda ml: ml.lot_name == lot_name or ml.lot_id.name == lot_name
```

Comment:
This line is the same as 329, package not used, is this correct?

Reply:
oops good spot